File: backend/src/main/java/com/maplewood/common/enums/EnrollmentStatus.java
package com.maplewood.common.enums;

/**
 * Enum for Current Enrollment Status
 * Maps to database: status IN ('enrolled', 'withdrawn')
 */
public enum EnrollmentStatus {
    ENROLLED("enrolled"),
    WITHDRAWN("withdrawn");

    private final String dbValue;

    EnrollmentStatus(String dbValue) {
        this.dbValue = dbValue;
    }

    public String getDbValue() {
        return dbValue;
    }

    public static EnrollmentStatus fromDbValue(String dbValue) {
        for (EnrollmentStatus status : EnrollmentStatus.values()) {
            if (status.dbValue.equalsIgnoreCase(dbValue)) {
                return status;
            }
        }
        throw new IllegalArgumentException("Invalid EnrollmentStatus: " + dbValue);
    }
}

File: frontend/src/types/EnrollmentStatus.ts
/**
 * Enrollment Status Enum
 * Matches backend EnrollmentStatus enum
 */
export const EnrollmentStatus = {
  ENROLLED: 'ENROLLED',
  WITHDRAWN: 'WITHDRAWN',
  COMPLETED: 'COMPLETED',
  FAILED: 'FAILED',
} as const;

export type EnrollmentStatus = typeof EnrollmentStatus[keyof typeof EnrollmentStatus];
